### PR TITLE
build: update scripts

### PR DESCRIPTION
**Pull request by ttytm**
The PR combines suggestions from recent discord messages. It should add pretty much universal coverage while still allowing to pass a custom compiler via the CC env var.

### CODE REVIEW

**Comment by ttytm** [comment]
Turns out TCC runs now into issues on Linux as well. Segfaults without clang / gcc. Maybe related to the segfaults users were mentioning in discord?

Run of last commit: https://github.com/ttytm/v-analyzer/actions/runs/7215726668/job/19660541723?pr=84

**Comment by ttytm** [comment]
Not possible for me to request a review, so I leave a mention @JalonSolov.

**Comment by JalonSolov** [comment]
Nice.  Looks better to me.  @spytheman ?

**Comment by ttytm** [comment]
Oh, I fucked up about the segfaults. TCC should be usable on linux, it's just that I added the static flag to all builds. It was just the release build before.

**Comment by ttytm** [comment]
We had it until now for linux dev and debug builds, and it apparently works. But please share all known friction points @Lycs-D.

File: build.vsh [hunk 1]
```diff
@@ -75,10 +84,9 @@ fn build(mode ReleaseMode, explicit_debug bool) {
 		println('Release mode is recommended for production use. At runtime, it is about 30-40% faster than debug mode.')
 	}
 
-	res := mode.compile()
```
Comment (spytheman) [inline build.vsh:78]: given this change, .compile() is not used anymore, and can be potentially removed.

**Review by spytheman — APPROVED** [review]

**Comment by spytheman** [comment]
> tcc can't compile tree-sitter

It still can not, but afaik only on Windows.

File: build.vsh (hunk 1 above)
Comment (ttytm) [reply build.vsh:78]: Right. Thanks!

**Comment by ttytm** [comment]
```
[ERROR] Failed to build v-analyzer
Building v-analyzer at commit: f[9](https://github.com/v-analyzer/v-analyzer/actions/runs/7221567521/job/19676795303#step:4:10)d5815, build time: 2023-12-15 11:40:47 ...
C:\hostedtoolcache\windows\v\0.4.3\x64\thirdparty\cJSON\cJSON.obj not found, building it (with msvc)...
✓ Prepared output directory
Building v-analyzer in release mode, using: C:\hostedtoolcache\windows\v\0.4.3\x64\v.exe ./cmd/v-analyzer -o ./bin/v-analyzer.exe -no-parallel -cc msvc  -prod
```

MSVC turns out to be problematic here atm.

**Comment by JalonSolov** [comment]
Yeah, msvc doesn't work for v-analyzer.  Use gcc on Windows to build it there.

**Pull request merged**